parlai/utils/misc.py

```diff
@@ -311,6 +311,21 @@ def __exit__(self, exc_type, exc_value, exc_traceback):
         pass
 
 
+class SimpleCounter:
+    """
+    Simple counter object.
+    """
+
+    def __init__(self, value=0):
+        self.val = value
+
+    def increment(self, value=1):
+        self.val += value
```

Comment:
I think this needs a lock :/

Reply:
I added a lock and it caused a deadlock actually. I got rid of it because I figured that the actual value doesn't matter as much as the fact that it increased in value

Reply:
I can try with a reentrant lock?